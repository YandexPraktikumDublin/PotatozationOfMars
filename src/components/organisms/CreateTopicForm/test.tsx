import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Store } from 'redux'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { CreateTopicForm } from '.'

const mockStore = configureStore([])

describe('<CreateTopicForm />', () => {
  let store: Store

  const initialState = {}

  beforeEach(() => {
    store = mockStore(initialState)
  })

  it('should renders correct <CreateTopicForm />', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <CreateTopicForm />
      </Provider>
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
