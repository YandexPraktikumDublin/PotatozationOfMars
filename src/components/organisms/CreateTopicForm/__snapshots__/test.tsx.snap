// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTopicForm /> should renders correct <CreateTopicForm /> 1`] = `
<BaseForm
  buttonText="Create"
  formError=""
  initialValues={
    Object {
      "title": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_excludedEdges": Array [],
      "_nodes": Array [
        "title",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "title": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "label": undefined,
            "meta": undefined,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "label": undefined,
        "meta": undefined,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <BaseInput
    name="subject"
    placeholder="Subject"
    type="text"
  />
  <BaseTextarea
    name="content"
    placeholder="Content"
  />
</BaseForm>
`;
